### PR TITLE
Emit Luis Trace from LUIS recognizer

### PR DESCRIPTION
**Pull request by emadelwany**

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs [hunk 1]
```diff
@@ -316,37 +327,55 @@ private static void AddProperties(LuisResult luis, RecognizerResult result)
...
 
+            var traceInfo = new LuisTraceInfo
+            {
+                RecognizerResult = recognizerResult,
+                Application = _application,
+                LuisResult = luisResult,
+                Options = _options,
+            };
+
+            await context.TraceActivityAsync("LuisRecognizerMiddleware", traceInfo, LuisTraceType, LuisTraceLabel, ct).ConfigureAwait(false);
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs]: LuisRecognizer

File: libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs [hunk 2]
```diff
@@ -20,38 +21,48 @@ namespace Microsoft.Bot.Builder.Ai.Luis
...
-            this.includeAPIResults = includeAPIResults;
+            _application = application;
+            _options = predictionOptions ?? new LuisPredictionOptions();
+            _includeApiResults = includeApiResults;
         }
 
         /// <inheritdoc />
-        public async Task<RecognizerResult> RecognizeAsync(string utterance, CancellationToken ct)
-            => await RecognizeInternalAsync(utterance, ct).ConfigureAwait(false);
+        public async Task<RecognizerResult> RecognizeAsync(TurnContext context, CancellationToken ct)
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs]: ITurnContext

File: libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs [hunk 3]
```diff
@@ -20,38 +21,48 @@ namespace Microsoft.Bot.Builder.Ai.Luis
     public class LuisRecognizer : IRecognizer
     {
         private const string MetadataKey = "$instance";
-        private readonly LuisRuntimeAPI runtime;
-        private readonly LuisApplication application;
-        private readonly LuisPredictionOptions options;
-        private readonly bool includeAPIResults;
+        private readonly LuisRuntimeAPI _runtime;
```
Comment (chrimc62) [inline libraries/Microsoft.Bot.Builder.Ai.LUIS/LuisRecognizer.cs]: Make sure there are no warnings.

**Comment by emadelwany** [comment]
@chrimc62 can you take a look? Please merge if it looks good to you

**Pull request merged**